### PR TITLE
feat(clerk-js): UserProfile form cleanup

### PR DESCRIPTION
**Pull request by desiprisg**
## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/clerk-js/src/ui/components/UserProfile/Web3Section.tsx [hunk 1]
```diff
@@ -3,10 +3,35 @@ import { useUser } from '@clerk/shared/react';
...
+    <RemoveWeb3WalletForm
+      onSuccess={close}
+      onReset={close}
+      {...props}
+    />
+  );
+};
+
+type Web3WalletScreenProps = { walletId?: string };
+const Web3WalletScreen = (props: Web3WalletScreenProps) => {
```
Comment (panteliselef) [inline packages/clerk-js/src/ui/components/UserProfile/Web3Section.tsx:25]: 🙃 Does this benefit us ?
```suggestion
const Web3WalletScreen = (props: ComponentProp<typeof AddWeb3Wallet>) => {
```

**Comment by octoper** [comment]
!preview

**Comment by clerk-cookie** [comment]
Hey @octoper, your preview is currently deploying and should be available shortly.

| Status | Preview | Updated (UTC) |
| :-- | :-- | :-- |
| :cook: _Deploying..._ | [Inspect](https://javascript-preview-olop368od.clerkstage.dev) | Dec 19, 2023 11:39 AM |

File: packages/clerk-js/src/ui/components/UserProfile/MfaTOTPScreen.tsx [hunk 2]
```diff
@@ -3,37 +3,30 @@ import React from 'react';
...
+        onReset={onReset}
       />
 
       <VerifyTOTP
-        onVerified={wizard.nextStep}
+        onSuccess={onSuccess}
+        onReset={onReset}
         resourceRef={ref}
       />
-
```
Comment (octoper) [inline packages/clerk-js/src/ui/components/UserProfile/MfaTOTPScreen.tsx:26]: We should keep the `<SuccessPage/>` here as we need to show to the user the backup codes.

**Review by octoper — CHANGES_REQUESTED** [review]

File: packages/clerk-js/src/ui/components/UserProfile/MfaPhoneCodeScreen.tsx [hunk 3]
```diff
@@ -39,34 +42,21 @@ export const MfaPhoneCodeScreen = withCardStateProvider(() => {
         title={localizationKeys('userProfile.mfaPhoneCodePage.title')}
         resourceRef={ref}
       />
-      <SuccessPage
```
Comment (octoper) [inline packages/clerk-js/src/ui/components/UserProfile/MfaPhoneCodeScreen.tsx:42]: We should keep the `<SuccessPage/>` here as we need to show to the user the backup codes.

**Comment by octoper** [comment]
!preview

**Review by octoper — APPROVED** [review]
💯

**Review by anagstef — APPROVED** [review]

**Pull request merged**